src/arch/mod.rs

```diff
@@ -0,0 +1,17 @@
+mod aarch64;
+mod manual;
+mod x86;
+
+#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+use std::is_x86_feature_detected;
+
+pub fn merge_impl() -> unsafe fn(crate::Rgba, crate::Rgba) -> crate::Rgba {
```

Comment:
Is it really needed to return a fn pointer?

Reply:
yeah we will see 3332